test/androidTest/java/org/thoughtcrime/securesms/ConversationListActivityIconsTest.java

```diff
@@ -44,78 +46,14 @@
 
     @Test
     public void conversationListActivityIconsTest() {
-
+        //opens overflow action bar
         openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
 
-        // Added a sleep statement to match the app's execution delay.
-        // The recommended way to handle such scenarios is to use Espresso idling resources:
-        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
-        try {
-            Thread.sleep(3000);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
+        //checks that all the actions have an icon
+        for(int i=0; i<6 ; i++) {
+            allOf(withId(R.id.icon), childAtPosition(childAtPosition(IsInstanceOf.<View>instanceOf(android.widget.ListView.class), i), 0), isDisplayed()).matches(isDisplayed());
```

Comment:
AllOf, has isDisplayed so checking if it's displayed is kind of redundant, also still kind of hard to read.